src/i18n.js

```diff
@@ -87,5 +97,15 @@ i18n
     () => setDocumentDir(i18n.language),
   )
 
+i18n.on('initialized', () => {
+  // HACK: Pass single language to changeLanguage to avoid wrong fallback behavior
+  // 'zh-TW' -> ['zh-hant', 'en'] instead of ['zh-TW', ...] -> ['zh-hans', 'en']
+  let language = i18n.services.languageDetector.detect()
```

Comment:
Does this still happen if the fallback is provided as a string?

Reply:
Not relevant. See https://github.com/i18next/i18next/issues/2299#issuecomment-2796673234. It is arguably a bug in `i18next.services.languageUtils.getBestMatchFromCodes` called only if array is provided by detector, wherein it checks for `zh-TW`, then base language `zh` (ignoring script), then first derivative (`zh-Hans` in our `supportedLng` case).